api/internal/crawl/doc/docname.go

```diff
@@ -53,6 +67,22 @@ func (doc *Document) FromRelativePath(newFile string) (Document, error) {
 }
 
 func (doc *Document) ID() string {
-	return doc.RepositoryURL + "/" +
-		doc.DefaultBranch + "/" + doc.FilePath
+	sum := sha256.Sum256([]byte(strings.Join(
+		[]string{
+			doc.RepositoryURL,
+			doc.DefaultBranch,
+			doc.FilePath,
+		},
+		"---|---")))
+	return fmt.Sprintf("%x", sum)
+}
+
+func (doc *Document) RepositoryFullName() string {
+	doc.RepositoryURL = strings.TrimRight(doc.RepositoryURL, "/")
+	sections := strings.Split(doc.RepositoryURL, "/")
```

Comment:
could this use path.Split instead? https://golang.org/pkg/path/#Split

Reply:
`strings.Split` is what we want here.